content/blogs/2023-10-09-why-ingestion-will-never-be-solved.md

```diff
@@ -0,0 +1,155 @@
+
+If you're an engineer trying to version-control your data integration configuration alongside your code, the no-code approach might not be the best fit for you.
+
+This is the niche that open-source tools such as [Singer](https://www.singer.io/), [Airbyte](https://airbyte.com/), [CloudQuery](https://www.cloudquery.io/) and [dltHub](https://dlthub.com/) are trying to fill.
+
+---
+
+## Open-source standards vs. open-source plugins
+
+When commercial no-code tools don't cut it, you'll likely "just" write a script. However, arbitrary ingestion scripts that don't adhere to any framework often lead to a lack of standardization and maintenance burden. There are two main approaches to solving this problem: open standards and open-source plugins.
+
+### Open standard: Singer
+
+[Stitch](https://www.stitchdata.com/) tried to solve that problem by building a universal data ingestion **standard** called [Singer spec](https://hub.meltano.com/singer/spec/). That spec aimed to improve standardization and community collaboration to avoid reinventing the wheel across companies.
+
+::alert{type="info"}
+To see how you can orchestrate Singer with [kestra](https://github.com/kestra-io/kestra), check the following [blueprint example](https://demo.kestra.io/ui/blueprints/community/70), and the [Singer plugin documentation](https://kestra.io/plugins/plugin-singer).
+::
+
+A typical [Singer](https://www.singer.io/) project consists of **Taps**, **Targets**, and a **JSON-based communication format** between them. Many connectors have been written using that specification. However, they were often developed by consulting firms in one-off projects, shifting the responsibility for long-term maintenance and continuous development to the end-user. As a result, the quality of those connectors varies greatly, and many claim that some Singer connectors are not production-ready.
+
+Especially after Stitch was [acquired](https://www.talend.com/about-us/press-releases/talend-acquires-stitch-a-leader-in-self-service-cloud-data-integration/) by Talend, there was no central entity governing the quality of connectors, leading to eroding trust in the data community.
+
+### Open-source connector ecosystems
+
+[Airbyte](https://airbyte.com/), [CloudQuery](https://www.cloudquery.io/) and [dltHub]([dltHub](https://dlthub.com/)) learned from the early successes and failures of [Singer](https://www.singer.io/). Instead of building a standard, all of them implemented their own **framework** and **platform** for building connectors using a pluggable system. This means that the open-source community can contribute to the connectors, and the platform vendor can ensure that those contributions are maintained and kept up-to-date.
```

Comment:
> failures of singer 
is maybe hard word ? even more that we have the plugin in Kestra

Reply:
it says: successes and failures :)

Reply:
I cut this now